src/com/google/enterprise/adaptor/filenet/DocumentTraverser.java

```diff
@@ -348,25 +368,23 @@ private void processInheritedPermissions(DocId docId,
 
   private Acl createAcl(DocId docId, Permissions.Acl permissions,
       PermissionSource permSrc, String parentFragment) {
-    Set<String> allowUsers = permissions.getAllowUsers(permSrc);
-    Set<String> denyUsers = permissions.getDenyUsers(permSrc);
-    Set<String> allowGroups = permissions.getAllowGroups(permSrc);
-    Set<String> denyGroups = permissions.getDenyGroups(permSrc);
-    if (allowUsers.isEmpty() && denyUsers.isEmpty()
-        && allowGroups.isEmpty() && denyGroups.isEmpty()) {
-      return null;
-    } else {
-      return createAcl(docId, parentFragment,
-          Acl.InheritanceType.CHILD_OVERRIDES,
-          options.getGlobalNamespace(), allowUsers, denyUsers,
-          allowGroups, denyGroups);
-    }
+    return createAcl(docId, parentFragment,
+        Acl.InheritanceType.CHILD_OVERRIDES,
```

Comment:
I think I would just have added the inheritance type as a parameter here and used it for all the inherited ACLs. If you just didn't want to touch the existing code, maybe do that in the followup, too?